src/internal/env-binding.c

```diff
@@ -40,5 +40,9 @@ sexp* rlang_env_get(sexp* env, sexp* nm) {
     r_abort("Internal error: `rlang_env_get()` should have failed earlier");
   }
 
+  if (r_typeof(out) == r_type_promise || r_typeof(out) == r_type_function) {
```

Comment:
Won't the second alternative evaluate bindings to a function? I think active bindings have promise type.

Reply:
Oooh good point; looks like I should be using this: https://github.com/wch/r-source/blob/496adb0085703362b2dee63537f0d89820f5afbb/src/include/Defn.h#L430

Reply:
But you're right that it is a subclass of promise

Reply:
Fixed now